hudi-utilities/src/main/java/org/apache/hudi/utilities/DeltaStreamerUtility.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities;
+
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.util.StringUtils;
+import org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer;
+import org.apache.hudi.utilities.deltastreamer.HoodieMultiTableDeltaStreamer;
+import org.apache.hudi.utilities.deltastreamer.TableExecutionContext;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class DeltaStreamerUtility {
+
+  public static String getDefaultConfigFilePath(String configFolder, String database, String currentTable) {
+    return configFolder + Constants.FILEDELIMITER + database + Constants.UNDERSCORE + currentTable + Constants.DEFAULT_CONFIG_FILE_NAME_SUFFIX;
+  }
+
+  public static String getTableWithDatabase(TableExecutionContext context) {
+    return context.getDatabase() + Constants.DELIMITER + context.getTableName();
+  }
+
+  public static void checkIfPropsFileAndConfigFolderExist(String commonPropsFile, String configFolder, FileSystem fs) throws IOException {
+    if (!fs.exists(new Path(commonPropsFile))) {
+      throw new IllegalArgumentException("Please provide valid common config file path!");
+    }
+
+    if (!fs.exists(new Path(configFolder))) {
+      fs.mkdirs(new Path(configFolder));
+    }
+  }
+
+  public static void checkIfTableConfigFileExists(String configFolder, FileSystem fs, String configFilePath) throws IOException {
+    if (!fs.exists(new Path(configFilePath)) || !fs.isFile(new Path(configFilePath))) {
+      throw new IllegalArgumentException("Please provide valid table config file path!");
+    }
+
+    Path path = new Path(configFilePath);
+    Path filePathInConfigFolder = new Path(configFolder, path.getName());
+    if (!fs.exists(filePathInConfigFolder)) {
+      FileUtil.copy(fs, path, fs, filePathInConfigFolder, false, fs.getConf());
+    }
+  }
+
+  public static TypedProperties getTablePropertiesFromConfigFolder(HoodieDeltaStreamer.Config cfg, FileSystem fs) throws IOException {
```

Comment:
@pratyakshsharma : I am rethinking about this feature again. Looking at the changes and added config options, I feel that we are overcomplicating in trying to reuse a config structure which for MultiDeltaStreamer  to fit to DeltaStreamer. If all we want to use HoodieDeltaStreamer but have config structure specific to HoodieMultiDeltaStreamer, can we just instantiate HoodieMultiDeltaStreamer with one source ?

Reply:
@bvaradar I guess you are right. My next question now is do we really want to reuse the config structure of HoodieMultiTableDeltaStreamer in HoodieDeltaStreamer or we can simply close this PR? 

If you strongly feel, I would do changes to instantiate HoodieMultiTableDeltaStreamer with one table, else please go ahead and close this PR :)

Reply:
@bvaradar got chance to decide for this? Its been there for some time now :)

Reply:
@pratyakshsharma : Sorry for the delay. I think we can close this PR. As a next step towards enhancing HoodieMultiDeltaStreamer, I think we can work on feature parity with : supporting parallel deltastreamer, support for MOR and async compaction. These would make HoodieMultiDeltaStreamer a really powerful tool :)